hack/generate-copyright.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+# Copyright The Shipwright Contributors
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+
+function listPkgDirs() {
+	go list -f '{{.Dir}}' ./cmd/... ./pkg/... ./test/... ./version/...
+  local goFiles=$?
+}
+
+function listGoFiles() {
+	# pipeline is much faster than for loop
+	listPkgDirs | xargs -I {} find {} -name '*.go'
+  local goFiles=$?
+  echo $PWD/tools.go
+  goFiles="$goFiles $?"
+}
+
+function listDockerfiles() {
+  find -name 'Dockerfile*' -not -path './vendor/*'
+}
+
+function listBashFiles() {
+  find -name '*.sh' -not -path './vendor/*'
+  local bashFiles=$?
+  find ./build/bin -type f
+  bashFiles="$bashFiles $?"
+}
+
+function listMarkdownFiles() {
+  find -name '*.md' -not -path './vendor/*'
+}
+
+function generateGoCopyright() {
+  allFiles=$(listGoFiles)
+
+  for file in $allFiles ; do
+    if ! head -n3 "${file}" | grep -Eq "(Copyright|SPDX-License-Identifier)" ; then
+      cp "${file}" "${file}.bak"
+      echo "// Copyright The Shipwright Contributors" > "${file}"
+      echo "// " >> "${file}"
+      echo "// SPDX-License-Identifier: Apache-2.0" >> "${file}"
+      echo "" >> "${file}"
+      cat "${file}.bak" >> "${file}"
+      rm "${file}.bak"
+    fi
+  done
+}
+
+function generateDockerfileCopyright() {
+  dockerfiles=$(listDockerfiles)
+  for file in $dockerfiles ; do
+    if ! head -n3 "${file}" | grep -Eq "(Copyright|SPDX-License-Identifier)" ; then
+      cp "${file}" "${file}.bak"
+      echo "# Copyright The Shipwright Contributors" > "${file}"
```

Comment:
I think it is better to define `Copyright The Shipwright Contributors` and `SPDX-License-Identifier: Apache-2.0` as parameter, or define the whole sentence as a shard function.

So that in future, if we want to refine or modify anything, we don't need to modify 3 places in this generate-copyright.sh script.

Reply:
I can parameterize the notice.

That said, changing the copyright on an open source project isn't a trivial matter. If hypothetically we relicense in the future, we may have to get permission from every contributor.

Reply:
@zhangtbj updated - ptal.